src/odemis/odemisd/modelgen.py

```diff
@@ -792,4 +857,54 @@ def get_instantiables(self, instantiated=None):
                 comps.add(n)
 
         return comps
+            
+    def read_yaml(self, f):
+        """
+        Read content of yaml file. This is a wrapper for yaml.safe_load that returns an
+        empty dictionary in case a YAMLError is raised or the file is empty.
+        f (File): path to yaml file
+        return (dict): dictionary with file contents or empty dictionary
+        """
+        try:
+            data = yaml.safe_load(f)
+            if not isinstance(data, dict):
+                data = {}
+        except yaml.YAMLError as ex:
+            logging.warning("Loading settings file failed with Exception %s" % ex)
```

Comment:
In the warning, it's good to indicate that the settings file is reset, and that Odemis will work fine.